addons/imaginary.sh

```diff
@@ -0,0 +1,144 @@
+    cat << IMAGINARY_DOCKER_CREATE > "$SCRIPTS"/imaginary-docker/docker-compose.yml
+version: '3.1'
+services:
+  imaginary:
+    image: nextcloud/aio-imaginary:latest
+    container_name: imaginary
+    restart: always
+    environment:
+       PORT: 9000
+    command: -concurrency 50 -enable-url-source -log-level debug
+    ports:
+      -  127.0.0.1:9000:9000
+IMAGINARY_DOCKER_CREATE
+    print_text_in_color "$IGreen" "SCRIPTS/imaginary-docker/docker-compose.yml was successfully created."
+fi
+
+# Start the container
+docker compose -p imaginary -f "$SCRIPTS"/imaginary-docker/docker-compose.yml up -d
+
+# Test if imaginary is working
+countdown "Testing if it works in 3 sedonds" "3"
+if curl -O "http://127.0.0.1:9000/crop?width=500&height=400&url=https://raw.githubusercontent.com/h2non/imaginary/master/testdata/large.jpg"
+then
+    print_text_in_color "$IGreen" "imaginary seems to be working OK!"
+else
+    msg_box "Test failed, please report this to: $ISSUES"
+    exit
+fi
+
+# Install preview generator just in case
+install_and_enable_app previewgenerator
+
+# check if the previewgenerator is installed and enabled
+if is_app_enabled previewgenerator
+then
+    # enable previews
+    nextcloud_occ config:system:set enable_previews --value=true --type=boolean
+
+    # install needed dependency for movies
+    install_if_not ffmpeg
+else
+    exit
+fi
+
+# Set providers
+nextcloud_occ config:system:set enabledPreviewProviders 0 --value="OC\\Preview\\JPEG"
+nextcloud_occ config:system:set enabledPreviewProviders 1 --value="OC\\Preview\\PNG"
+nextcloud_occ config:system:set enabledPreviewProviders 2 --value="OC\\Preview\\WEBP"
+nextcloud_occ config:system:set enabledPreviewProviders 3 --value="OC\\Preview\\HEIF"
+nextcloud_occ config:system:set enabledPreviewProviders 4 --value="OC\\Preview\\TIFF"
+nextcloud_occ config:system:set enabledPreviewProviders 5 --value="OC\\Preview\\PDF"
+nextcloud_occ config:system:set enabledPreviewProviders 6 --value="OC\\Preview\\GIF"
+nextcloud_occ config:system:set enabledPreviewProviders 7 --value="OC\\Preview\\SVG"
+nextcloud_occ config:system:set enabledPreviewProviders 8 --value="OC\\Preview\\MP3"
+nextcloud_occ config:system:set enabledPreviewProviders 9 --value="OC\\Preview\\TXT"
+nextcloud_occ config:system:set enabledPreviewProviders 10 --value="OC\\Preview\\MarkDown"
+nextcloud_occ config:system:set enabledPreviewProviders 11 --value="OC\\Preview\\OpenDocument"
+nextcloud_occ config:system:set enabledPreviewProviders 12 --value="OC\\Preview\\Krita"
+nextcloud_occ config:system:set enabledPreviewProviders 13 --value="OC\\Preview\\BMP"
+nextcloud_occ config:system:set enabledPreviewProviders 14 --value="OC\\Preview\\Movie"
+nextcloud_occ config:system:set enabledPreviewProviders 15 --value="OC\\Preview\\Imaginary"
+nextcloud_occ config:system:set preview_imaginary_url --value="http://127.0.0.1:9000"
+
+# Set values
+nextcloud_occ config:app:set previewgenerator squareSizes --value="32 256"
```

Comment:
This is not taken into account by imaginary afaik

Reply:
Same with heightsize

Reply:
Ok, thanks!